test/handlers.test.js

```diff
@@ -128,7 +214,26 @@ describe(__filename, function () {
         expect(stubs['./http'].getCall(0).args[0]).to.deep.equal({
           guid: guid,
           method: 'DELETE',
-          path: '/' + dataset + '/' + id,
+          path: urlJoin(pathToPrepend, dataset, id),
+          timeout: 25000
+        });
+
+        done();
+      });
+    });
+  });
+
+  describe('#handleDelete', function () {
```

Comment:
Thanks for updating these tests. I think the new structure is unusual though. We have multiple _describe_ calls for the same functions. Instead, could you use multiple _it_ calls?

For example:

```js
describe('#handleDelete', function () {
  it('should work without a nested path', function (done) { /*etc. */});

  it('should work with a nested path', function (done) { /*etc. */})
});
```

I also think you only need one new test case for the path handling if you abstract it into a small module rather than making a new test for each function? It would also provide a cleaner API since you could just omit _pathToPrepend_ and not need to pass an empty string.

Example:

```js

module.exports = function buildRoute (opts, dataset) {
  return (opts.pathToPrepend) ? (opts.pathToPrepend + dataset) :  dataset;
};

```